packages/web-components/fast-foundation/src/foundation-element/foundation-element.ts

```diff
@@ -216,7 +221,7 @@ export class FoundationElementRegistry<
         const prefix = definition.prefix || context.elementPrefix;
         const name = `${prefix}-${definition.baseName}`;
 
-        context.tryDefineElement(name, this.type, x => {
+        context.tryDefineElement(name, overrideDefinition.baseClass ?? this.type, x => {
```

Comment:
I *think* that passing the base class here instead of the actual class may cause some problems with the platform element definition. What if we allow an optional third argument to `tryDefineElement` that takes an `options` object that allows passing in an additional type for name mapping?

Reply:
I think you're right, and either way it would be a lot safer to implement, I'll add it! 👍

Reply:
@EisenbergEffect I ended up breaking the parameter list for an object for scalability, let me know what you think about it. I also chose to register the actual class _and_ the baseClass (if any) for better flexibility. I could really use your review at this point just to make sure what I'm doing still makes sense 😅